storage_controller/src/service.rs

```diff
@@ -3958,6 +3991,9 @@ impl Service {
         let import_failed = import.completion_error().is_some();
 
         if !import_failed {
+            self.reset_tenant_post_import(&import).await?;
```

Comment:
What if the import failed? We should clean up by deleting the timeline.
Can we even do that in the state it's in? Probably not.
Do we have a ticket for that? Would have liked to see an

```rust
} else {
    // TODO: link to ticket
    warn!("leaking timeline that failed to import");
}

Reply:
As discussed, it's in the next PR.